### PR TITLE
Centralize viewArea state in Visualization

### PR DESCRIPTION
**Pull request by thequilo**
The viewAreas are now managed in a central location instead of in the local minimap objects. The central state is a single source of truth and eliminates much of the communication between plot objects. This has a few benefits:
- much shorter implementations for the minimaps (because the communication code is gone)
- reasoning about state changes is easier because everything is handled in a central location
- throttling is synchronized between plots and applied to both word filtering and plotting
- It is easier to determine which plots have to be updated, which results in significant speedups for viewing long recordings 

Additionally, functionality was added to synchronize plots in a side-by-side view.

### CODE REVIEW

File: meeteval/viz/visualize.js [hunk 1]
```diff
@@ -23,6 +23,169 @@ var colormaps = {
...
+/**
+ * Checks whether two ranges `a` and `b` are similar by relative tolerance `tolerance`.
+ */
+function similar_range(a, b, tolerance=0.00001){
+    // Note: a == [0, 1] will always return false, because arrays are compared by reference, not values
+    if (JSON.stringify(a) === JSON.stringify(b)) {
+        return true
+    }
+
+    if (!a || a.some(isNaN) || a.some(isNaN))
```
Comment (boeddeker) [inline meeteval/viz/visualize.js:107]: Why do you test only a?

Comment (boeddeker) [reply meeteval/viz/visualize.js:107]: a.some(isNaN) is repeated

File: meeteval/viz/visualize.js [hunk 2]
```diff
@@ -23,6 +23,169 @@ var colormaps = {
...
+    if (viewArea) {
+        viewAreas[anchor] = viewArea;
+        dirty[anchor] = true;
+    }
+
+    // Update children of anchor (j > i)
+    for (let j = anchor + 1; j < viewAreas.length; j++) {
+        let v = viewAreas[j];
+        const parent = viewAreas[j - 1];
+        if (parent[1] - parent[0] < v[1] - v[0] || similar_range(oldViewAreas[j - 1], v)) {
```
Comment (boeddeker) [inline meeteval/viz/visualize.js:144]: Should it be marked as dirty, when the values ares the same? A draw wouldn't be necessary.

File: meeteval/viz/visualize.js [hunk 3]
```diff
@@ -23,6 +23,169 @@ var colormaps = {
...
+ * Ensures that every entry (index i) lies within its parent (index i - 1).
+ *
+ * Keeps the viewArea at `anchor` fixed.
+ *
+ * Returns an array of booleans indicating which viewAreas were changed.
+ */
+function adjustViewAreas(viewAreas, anchor, viewArea) {
+    const dirty = viewAreas.map(() => false);
+
+    const oldViewAreas = [...viewAreas];
```
Comment (boeddeker) [inline meeteval/viz/visualize.js:133]: I am not sure, if it works as intended. You do inplace changes and keep only the old value if the child is larger than the parent.
If this is the  indented behavior, it should be documented.

File: meeteval/viz/visualize.js [hunk 4]
```diff
@@ -23,6 +23,169 @@ var colormaps = {
...
+    for (let j = anchor - 1; j > 0; j--) {
+        let v = viewAreas[j];
+        const child = viewAreas[j + 1];
+        if (child[1] - child[0] > v[1] - v[0]) {
+            v = child;
+            dirty[j] = true;
+        } else if (v[0] > child[0]) {
+            const diff = child[0] - v[0];
+            viewAreas[j][0] += diff;
+            viewAreas[j][1] += diff;
```
Comment (boeddeker) [inline meeteval/viz/visualize.js:175]: change v?

File: meeteval/viz/visualize.js (hunk 1 above)
Comment (thequilo) [reply meeteval/viz/visualize.js:107]: Done

File: meeteval/viz/visualize.js (hunk 2 above)
Comment (thequilo) [reply meeteval/viz/visualize.js:150]: Solved, now only marked as dirty when `!similar_rate(viewAreas[j], v)`

**Pull request merged**